python_modules/libraries/dagster-databricks/dagster_databricks/databricks_step_main.py

```diff
@@ -102,6 +102,23 @@ def setup_storage(step_run_ref):
         raise Exception('No valid storage found!')
 
 
+def setup_environment(step_run_ref):
+    '''Setup any environment variables required by the run.
+
+    Extract any secrets in the run config and export them as environment variables.
+
+    This is important for any `StringSource` config since the environment variables
+    won't ordinarily be available in the Databricks execution environment.
+    '''
+    for secret in step_run_ref.run_config.get('secrets', []):
```

Comment:
This would need to be `step_run_ref.run_config['resources'][databricks_resource_name].get('secrets')`, right?  And in theory there could be multiple databricks resources?

Reply:
Good spot. I'd glossed over the fact that the name `pyspark_step_launcher` is chosen by the user rather than the library; that will need improving in the storage setup as well. Let's discuss in the main comment thread, since I agree we shouldn't be touching `run_config` like this here!